site/src/Consumer/EmailNotificationConsumer.php

```diff
@@ -12,26 +14,105 @@
  */
 class EmailNotificationConsumer extends AbstractNotificationConsumer
 {
+    /**
+     * @var UserManagerInterface
+     */
+    protected $userManager;
+
+    /**
+     * @var \Swift_Mailer
+     */
+    protected $mailer;
+
+    /**
+     * @var Environment
+     */
+    protected $twigEnvironment;
+
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
     /**
      * EmailNotificationConsumer constructor
      *
-     * @param LoggerInterface $logger
+     * @param LoggerInterface      $logger
+     * @param UserManagerInterface $userManager
+     * @param Environment          $twigEnvironment
+     * @param \Swift_Mailer        $mailer
+     * @param ContainerInterface   $container
      */
-    public function __construct(LoggerInterface $logger)
-    {
+    public function __construct(
+        LoggerInterface $logger,
+        UserManagerInterface $userManager,
+        Environment $twigEnvironment,
+        ContainerInterface $container
+    ) {
         parent::__construct($logger);
+        $this->userManager = $userManager;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->container = $container;
+        $this->mailer = $container->get(
+            'swiftmailer.mailer.notification_mailer'
+        );
+    }
+
+    protected function shouldHandle()
+    {
+        $type = $this->notification->getType();
+
+        $user = $this->userManager->findUserByUsername($this->notification->getRecipient()->getUsername());
+
+        $emailSubscription = $user->isSubscribedToEmailNotifications();
+
+        if ($emailSubscription) {
+            switch ($type) {
+                case "site_information":
+                    $this->logger->info('Processing the notification and sending an email...');
+
+                    return true;
+                default:
+                    $this->logger->error('Something went wrong. Please check the notification type');
+
+                    return false;
+            }
+        }
```

Comment:
Remove this switch/case. We don't need to differentiate between types, we send out all of them as email.

Reply:
We do! Deleted the switch case